plugins/es-repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -119,11 +121,19 @@ static ClientConfiguration buildConfiguration(S3ClientSettings clientSettings) {
     }
 
     // pkg private for tests
-    static AWSCredentialsProvider buildCredentials(Logger logger, S3ClientSettings clientSettings) {
+    static AWSCredentialsProvider buildCredentials(Logger logger, S3ClientSettings clientSettings, String repository) {
         final AWSCredentials credentials = clientSettings.credentials;
         if (credentials == null) {
             logger.debug("Using instance profile credentials");
-            return new EC2ContainerCredentialsProviderWrapper();
+            var ec2ContainerCredentialsProviderWrapper = new EC2ContainerCredentialsProviderWrapper();
+            try {
+                // Try to connect to test credentials
+                ec2ContainerCredentialsProviderWrapper.getCredentials();
+                return ec2ContainerCredentialsProviderWrapper;
+            } catch(SdkClientException e) {
+                throw new RepositoryVerificationException(repository, "Cannot find required credentials to create a repository of type s3. " +
```

Comment:
I am just wondering why it should `RepositoryVerificationException`, see just trying to configure a client at this point and not yet doing anything with repositories. Also to `buildCredentials` we don't need to know the repository name ` buildCredentials(Logger logger, S3ClientSettings clientSettings, String repository)`

Reply:
Any better idea ? IOException ?

Reply:
but why `RepositoryVerificationException` if it has nothing to do with repositories?

Reply:
This class is part of the repository package.